src/two_party/lindell_2017/party2.rs

```diff
@@ -191,8 +191,9 @@ impl MasterKey2 {
             party_two::PaillierPublic,
             Challenge,
             VerificationAid,
+            party_two::PDLchallenge,
         ),
-        ProofError,
+        (),
```

Comment:
Why removing the ProofError?

Reply:
proof error is too general naming. In this PR I played a bit with the way we propagate errors . Most of the cases, using `expect` is the best alternatives. In newer designs I have a different method: I define Error Enum in `lib.rs` and use it to specify the type of error I have in the lib. It is the lib job to translate errors coming from different libraries to the errors in the enum

Reply:
Ok. can you add a small comment in the code explaining why do we return a "generic" error?